guacamole-ext/src/main/java/org/apache/guacamole/net/auth/AbstractUser.java

```diff
@@ -44,4 +51,164 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * <p>This implementation simply an immutable, empty map. Implementations
+     * that wish to expose custom attributes should override this function.
+     */
+    @Override
+    public Map<String, String> getAttributes() {
+        return Collections.<String, String>emptyMap();
```

Comment:
Are these type qualifiers actually needed?

Reply:
I think it can be inferred.